app/browser/reducers/windowsReducer.js

```diff
@@ -4,18 +4,276 @@
 
 'use strict'
 
+const path = require('path')
 const appConstants = require('../../../js/constants/appConstants')
 const windowConstants = require('../../../js/constants/windowConstants')
 const windowState = require('../../common/state/windowState')
 const windows = require('../windows')
 const {makeImmutable} = require('../../common/state/immutableUtil')
+const electron = require('electron')
+const BrowserWindow = electron.BrowserWindow
+const firstDefinedValue = require('../../../js/lib/functional').firstDefinedValue
+const appConfig = require('../../../js/constants/appConfig')
+const messages = require('../../../js/constants/messages')
+const appUrlUtil = require('../../../js/lib/appUrlUtil')
+const settings = require('../../../js/constants/settings')
+const getSetting = require('../../../js/settings').getSetting
```

Comment:
same thing for all of these, this is just copied over and will go through a second round of refactoring to make other changes